### ISSUE TITLE
Edit link for logged in user on dataset page

### ISSUE THREAD

**Issue by cmugisha** [GetDKAN/data-catalog-app#74]
On DKAN1, users were accustomed to going to a dataset page and seeing an edit button they could use make changes. Currently, to edit a dataset, users need to go to a manage content page in order to find and edit link to datasets. In user tests testing, this was found to be confusing for users who were accustomed to DKAN1.

* User story*
As a data publisher, when I am looking at a dataset page, I would like to be able to choose to edit it from that page, so that I don't need to leave the page to make changes.

*Acceptance criteria*
- Tests and code for 1.x and 2.x to render an edit link conditionally on the datasets page.

*Estimate*
8 story points.

* References *
This work builds on this spike ticket: https://github.com/GetDKAN/data-catalog-app/issues/65

**Comment by alexiscott** [comment]
Discussed the implementation approach with @dgading for global state.
Wrapping `<UserContext><Route>..`. should  make it possible to go ahead and the User logged in state from within _any_ component with `useContext(UserContext).  That strategy frees me up to start working on this.

**Comment by alexiscott** [comment]
@cmugisha just a heads up that I have assigned the PR for this ticket to Daniel Gading. Once it is merged and released we will be able to add the component and tests that will provide the link to the logged in user's dataset in Drupal.  I don't think that we will need a new ticket for that work.

**Comment by cmugisha** [comment]
@dgading Any progress in reviewing this?

**Comment by cmugisha** [comment]
Slack comment from Daniel Gading: I talked to Alex about it. This PR will be closed and the work should be done on the individual catalog apps for now. If I merged this into data-catalog-app it would still require refactors on each of the site catalogs since this code isn't pulled in but once when the site is initially built. I think it is a good start, but should be moved to a custom hook or something in a different repo so it is a little more reuseable, but with our new time tables, that isn't really feasible right now

**Comment by cmugisha** [comment]
Chadia's Slack response to Daniel: Ok. It  sounds like this ticket needs to be moved back to In Progress or should I close it so that you and Alex can create new tickets that better reflect the individual work you just mentioned?

**Comment by cmugisha** [comment]
@alexiscott See conversation between Daniel and I above and advise on how to proceed with this ticket.

**Comment by cmugisha** [comment]
Response from Daniel: I think this can be closed. And open one in each repo that we are migrating that needs it
